src/mqttproto/client_state_machine.py

```diff
@@ -37,6 +37,7 @@ class MQTTClientStateMachine(BaseMQTTClientStateMachine):
         validator=instance_of(str), factory=lambda: f"mqttproto-{uuid4().hex}"
     )
     _ping_pending: bool = field(init=False, default=False)
+    may_retain: bool = field(init=False, default=True)
```

Comment:
If we're going to expose a bunch of broker capabilities, I think it would be best to encapsulate them in a separate class (`BrokerCapabilities`).

Reply:
Makes sense. Let's refactor to do that once the bits that are currently spread among three different PRs are merged into one.

Reply:
At the very least make this a read-only property, just like `maximum_qos`.

Reply:
OK